diff --git a/api/script-recorded-program-file.vm.js b/api/script-recorded-program-file.vm.js
index 2822efb..db81c62 100644
--- a/api/script-recorded-program-file.vm.js
+++ b/api/script-recorded-program-file.vm.js
@@ -14,7 +14,7 @@
 			
 			if (request.type === 'm2ts') {
 				response.setHeader('content-length', fstat.size);
-				response.setHeader('content-disposition', 'attachment; filename="' + program.id + '.m2ts"');
+				response.setHeader('content-disposition', 'attachment; filename="' + program.id + program.recorded.match(/\.[^\.]+?$/)[0] + '"; filename*=UTF-8\'\'' +  encodeURI(program.recorded.match(/\/[^\/]+?$/)[0].substr(1)));
 				response.head(200);
 
 				fs.createReadStream(program.recorded).pipe(response);
diff --git a/api/script-recorded-program-preview.vm.js b/api/script-recorded-program-preview.vm.js
index b6a8200..59856dd 100644
--- a/api/script-recorded-program-preview.vm.js
+++ b/api/script-recorded-program-preview.vm.js
@@ -44,7 +44,7 @@
 	
 	var ffmpeg = child_process.exec(
 		(
-			'ffmpeg -f mpegts -ss ' + pos + ' -r 10 -i "' + program.recorded + '" -ss 1.5 -r 10 -frames:v 1' +
+			'ffmpeg -ss ' + pos + ' -r 10 -i "' + program.recorded + '" -ss 1.5 -r 10 -frames:v 1' +
 			' -c:v ' + vcodec + ' -an -f image2 -s ' + width + 'x' + height + ' -map 0:0 -y pipe:1'
 		)
 		,
diff --git a/api/script-recorded-program-watch.vm.js b/api/script-recorded-program-watch.vm.js
index 5bc2156..5090418 100644
--- a/api/script-recorded-program-watch.vm.js
+++ b/api/script-recorded-program-watch.vm.js
@@ -190,7 +190,7 @@ function main(avinfo) {
 				args.push("-hwaccel_output_format", "yuv420p");
 			}
 
-			args.push('-i', 'pipe:0');
+			args.push('-i', program.recorded);
 
 			if (d.t) { args.push('-t', d.t); }
 
@@ -262,13 +262,11 @@ function main(avinfo) {
 
 			args.push('-y', '-f', d.f, 'pipe:1');
 
-			var readStream = fs.createReadStream(program.recorded, range || {});
-
-			request.on('close', function() {
-				readStream.destroy();
-			});
-
 			if (d['c:v'] === 'copy' && d['c:a'] === 'copy' && !d.t) {
+				var readStream = fs.createReadStream(program.recorded, range || {});
+				request.on('close', function() {
+					readStream.destroy();
+				});
 				readStream.pipe(response);
 			} else {
 				var ffmpeg = child_process.spawn('ffmpeg', args);
@@ -277,8 +275,6 @@ function main(avinfo) {
 
 				ffmpeg.stdout.pipe(response);
 
-				readStream.pipe(ffmpeg.stdin);
-
 				ffmpeg.stderr.on('data', function(d) {
 					util.log('#ffmpeg: ' + d);
 				});
